crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -726,6 +728,20 @@ impl<'this, TX: DbTx<'this>> AccountExtReader for DatabaseProvider<'this, TX> {
     }
 }
 
+impl<'this, TX: DbTx<'this>> AccountChangeReader for DatabaseProvider<'this, TX> {
+    fn account_changes_in_block(&self, block_number: BlockNumber) -> Result<Vec<AccountBeforeTx>> {
+        let range = block_number..=block_number;
+        self.tx.cursor_read::<tables::AccountChangeSet>()?.walk_range(range)?.try_fold(
```

Comment:
I think you can use `.walk_range(range)?.map(|res| res.map(|(_,val)| val)).collect::<Result<Vec<_>>>()`

Reply:
Ah got it, I was using `try_fold` since it seems like walk_range's `RangeWalker.Item = Result<_, DatabaseError>` so iteration can fail w/ a DB error and I was naively reaching for a `try_map` w/o finding one.  

Again that's just me not knowing rust, but based on your suggestion I found https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect and agree that's more idiomatic, thanks!

Reply:
Also in 80810b804ebaea75aa500e1673c9c54bd2cc8b37, I went with a slightly more verbose option but as mentioned at https://github.com/paradigmxyz/reth/pull/3768#discussion_r1264455758 I'm ok making this more terse if that's the preferred style.